rockcraft/project.py

```diff
@@ -180,10 +181,33 @@ class Service(pydantic.BaseModel):
     kill_delay: Optional[str] = pydantic.Field(alias="kill-delay")
 
 
+NAME_REGEX = r"^([a-z](?:-?[a-z0-9]){2,})$"
```

Comment:
for future reference, this is the regular expression for the OCI-compliant image name and tag (the latter could be applied to rockcraft.yaml's `version` field): https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pulling-manifests